gems/pending/util/extensions/miq-file.rb

```diff
@@ -52,7 +53,7 @@ def self.sizeEx(path)
 
   def self.path_to_uri(file, hostname = nil)
     hostname ||= MiqSockUtil.getFullyQualifiedDomainName
-    URI.join("file://#{hostname}", "/#{URI.encode(file.tr('\\', '/'))}").to_s
+    URI.join("file://#{hostname}", "/#{Addressable::URI.encode(file.tr('\\', '/'))}").to_s
```

Comment:
I guess it couldn't hurt to add support for that. I'll update the PR and add another spec.